File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode dummy = new ListNode(0);
        ListNode tail = dummy;
        ListNode head1 = list1;
        ListNode head2 = list2 ;

        while(head1 != null && head2 != null){
             
            if(head1.val <= head2.val){
                tail.next = head1;
                tail = tail.next ;
                head1 = head1.next;
            }
            else{
                tail.next = head2;
                tail = tail.next ;
                head2 = head2.next;
            }


        }

        while(head1 != null || head2 != null){
            if(head1 != null ){
                tail.next = head1;
                tail = tail.next ;
                head1 = head1.next;
            }
            else{
                tail.next = head2;
                tail = tail.next ;
                head2 = head2.next;
            }
        }

       return dummy.next ; 
        
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def mergeTwoLists(self, list1, list2):
        h1 = list1
        h2 = list2
        dummy = ListNode(0)
        curr = dummy

        while h1 and h2:
            if h1.val >= h2.val :
                curr.next = h2
                h2 = h2.next
            else:
                curr.next = h1
                h1 = h1.next    
            
            curr = curr.next

        if h1:
            curr.next = h1
        else:
            curr.next = h2    

        return dummy.next
        